tools/ddprobe.cpp

```diff
@@ -69,9 +69,131 @@ syncThreadDesktop() {
   CloseDesktop(hDesk);
 }
 
+/**
+ * @brief Determines whether a given frame is entirely black by checking every pixel.
+ *
+ * This function checks if the provided frame is entirely black by inspecting each pixel in both the x and y dimensions. It inspects the RGB channels of each pixel and compares them against a specified black threshold. If any pixel's RGB values exceed this threshold, the frame is considered not black, and the function returns `false`. Otherwise, if all pixels are below the threshold, the function returns `true`.
+ *
+ * @param mappedResource A reference to a `D3D11_MAPPED_SUBRESOURCE` structure that contains the mapped subresource data of the frame to be analyzed.
+ * @param frameDesc A reference to a `D3D11_TEXTURE2D_DESC` structure that describes the texture properties, including width and height.
+ * @param blackThreshold A floating-point value representing the threshold above which a pixel's RGB channels are considered non-black. The value ranges from 0.0f to 1.0f, with a default value of 0.1f.
+ * @return bool Returns `true` if the frame is determined to be black, otherwise returns `false`.
+ */
+bool
+isFrameBlack(const D3D11_MAPPED_SUBRESOURCE &mappedResource, const D3D11_TEXTURE2D_DESC &frameDesc, float blackThreshold = 0.1f) {
```

Comment:
Sunshine uses snake_case naming conventions. I think newly added code should also use it.

Reply:
It's a mixture of both actually, the only consistent behavior was that it appeared more internal functions weren't snake_cased so that's why I did it that way, fixing it is no big deal either way

Reply:
We have a lot of inconsistency. The new pattern would ideally be snake case though.

While I like the idea of camel case... it has some issues. For example take a function named `get_gfe_version`... in camel case, that's `getGfeVersion` or `getGFEVersion`? Both of those are harder to read and I'm not sure either is correct.

Mostly, I like to follow what python does with PEP8, as it's pretty well thought out and defined.